salesagility/SuiteCRM
# Fix function declarations of static functions in SubPanelDefinitions

## Description




Declare the functions SubPanelDefinitions::get_all_subpanels(),
SubPanelDefinitions::set_hidden_subpanels() and SubPanelDefinitions::get_hidden_subpanels()
as static functions since they are intended to be used as such.

## Motivation and Context

Statically calling non-static functions is deprecated.

## How To Test This

1. In `php.ini` change the `error_reporting` setting from something like
```
error_reporting = E_ALL & ~E_DEPRECATED & ~E_STRICT
```
to
```
error_reporting = E_ALL & ~E_STRICT
```
2. Restart Apache (e.g. `service apache2 reload`) or PHP-FPM (e.g. `service php7.1-fpm restart`)
3. In SuiteCRM's Admin panel navigate to "Display Modules and Subpanels"
4. Check Apache error log for messages like
```
PHP Deprecated:  Non-static method SubPanelDefinitions::get_all_subpanels()
should not be called statically in
/var/www/html/suitecrm/modules/Administration/views/view.configuretabs.php on line 109
```
After applying this PR this message should not appear anymore.

## Types of changes


### Final checklist
